File: 1716-calculate-money-in-leetcode-bank/1716-calculate-money-in-leetcode-bank.js
const totalMoney = (n) => {
  let i = 0, sum = 0;
  while (i < 7) {
    const d = Math.floor(n / 7) + i + (n % 7 > i ? 1 : 0);
    sum += d * (d + 1) / 2 - i * (i + 1) / 2;
    ++i;
  }
  return sum;
};

File: 1716-calculate-money-in-leetcode-bank/1716-calculate-money-in-leetcode-bank.go
func totalMoney(n int) int {
  ans := 0
  for add, i := 1, 1; i <= n; i += 1 {
    ans += add
    add += 1
    if i % 7 == 0 {
      add = add - 7 + 1
    }
  }
  return ans
}